labelbox/data/metrics/iou.py

```diff
@@ -0,0 +1,290 @@
+        labels: List of masks objects
+    Returns:
+        float indicating iou score
+    """
+
+    pred_mask = _instance_urls_to_binary_mask(
+        [pred.mask['instanceURI'] for pred in predictions])
+    label_mask = _instance_urls_to_binary_mask(
+        [label.mask['instanceURI'] for label in labels])
+    assert label_mask.shape == pred_mask.shape
+    return _mask_iou(label_mask, pred_mask)
+
+
+def classification_iou(predictions: List[ClassificationTool],
+                       labels: List[ClassificationTool]) -> float:
+    """
+    Computes iou for classification features.
+
+    Args:
+        prediction : list of predictions for a particular feature schema ( should have a max of one ).
+        label : list of predictions for a particular feature schema ( should have a max of one ).
+    Returns:
+        float indicating iou score.
+
+    """
+
+    if len(predictions) != len(labels) != 1:
+        return 0.
+
+    if len(predictions) == 1:
+        prediction = predictions[0]
+    if len(labels) == 1:
+        label = labels[0]
+
+    if type(prediction) != type(label):
+        return 0.
+
+    if isinstance(prediction, NDText):
+        return float(prediction.answer == label.answer)
+    elif isinstance(prediction, NDRadio):
+        return float(prediction.answer.schemaId == label.answer.schemaId)
+    elif isinstance(prediction, NDChecklist):
+        schema_ids_pred = {answer.schemaId for answer in prediction.answers}
+        schema_ids_label = {answer.schemaId for answer in label.answers}
+        return float(
+            len(schema_ids_label & schema_ids_pred) /
+            len(schema_ids_label | schema_ids_pred))
+    else:
+        raise ValueError(f"Unexpected subclass. {prediction}")
+
+
+def subclassification_iou(
+        subclass_predictions: List[ClassificationTool],
+        subclass_labels: List[ClassificationTool]) -> Optional[float]:
+    """
+
+    Computes subclass iou score between two vector tools that were matched.
+
+    Arg:
+        subclass_predictions: All subclasses for a particular vector feature inference
+        subclass_labels : All subclass labels for a label that matched with the vector feature inference.
+
+    Returns:
+        miou across all subclasses.
+    """
+
+    subclass_predictions = create_schema_lookup(subclass_predictions)
+    subclass_labels = create_schema_lookup(subclass_labels)
+    feature_schemas = set(subclass_predictions.keys()).union(
+        set(subclass_labels.keys()))
+    # There should only be one feature schema per subclass.
+
+    classification_iou = [
+        feature_miou(subclass_predictions[feature_schema],
+                     subclass_labels[feature_schema])
+        for feature_schema in feature_schemas
+    ]
+    classification_iou = [x for x in classification_iou if x is not None]
+    return None if not len(classification_iou) else np.mean(classification_iou)
+
+
+def get_vector_pairs(predictions: List[Dict[str, Any]], labels):
```

Comment:
Maybe prefix this function name with underscore (and other functions in this file that are internal only, like `preprocess_args`)